### PR TITLE
Ops/Build Additions (Makefile/Vendoring/Docker)

### PR DESCRIPTION
**Pull request by chiefy**
* Add `Makefile`
  * Add `Dockerfile` w/ texlive setup file
  * Add vendoring w/ `dep`
  * Add sample `docker-compose.yml`
  * Update `README.md` with instructions on how to build docker etc.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+TARGET:=$(GOPATH)/bin/grafana-reporter
+SRC:=$(GOPATH)/src/github.com/IzakMarais/reporter
+
+.PHONY: build
+build: $(TARGET)
+
+.PHONY: docker-build
+docker-build:
+	@docker build -t grafana-reporter:0.6.2 .
```
Comment (IzakMarais) [inline Makefile]: Where does this version number 0.6.2. come from?

I should really start tagging releases and doing proper semver for this project....

Comment (chiefy) [reply Makefile]: @IzakMarais was just using `version.go` https://github.com/IzakMarais/reporter/blob/master/cmd/grafana-reporter/version.go#L20-L22

Yeah, it'd be better if you used an `.env` file or something to read from that way build tools can get at the version easier.

**Comment by chiefy** [comment]
@IzakMarais the problem is you're going to run into issues w/ travis CI:

from the build logs:
```
$ mkdir -p $HOME/gopath/src/github.com/IzakMarais/reporter
```

**Comment by chiefy** [comment]
after changes:
```
~/GitHub/go/src/github.com/izakmarais/reporter chiefy-master 21s
❯ make docker-build
Sending build context to Docker daemon  12.14MB
Step 1/10 : FROM golang:1.8-stretch AS build
 ---> feba821534bd
Step 2/10 : WORKDIR /go/src/${owner:-github.com/izakmarais}/reporter
Removing intermediate container 1781b8885dbb
 ---> d734f4d48fb9
Step 3/10 : RUN apt-get update && apt-get install make git
 ---> Running in 17fdae09cc87
Ign:1 http://deb.debian.org/debian stretch InRelease
Get:2 http://deb.debian.org/debian stretch-updates InRelease [91.0 kB]
Get:3 http://security.debian.org stretch/updates InRelease [63.0 kB]
Get:4 http://deb.debian.org/debian stretch Release [118 kB]
Get:5 http://deb.debian.org/debian stretch-updates/main amd64 Packages [8431 B]
Get:6 http://deb.debian.org/debian stretch Release.gpg [2434 B]
Get:7 http://deb.debian.org/debian stretch/main amd64 Packages [9531 kB]
Get:8 http://security.debian.org stretch/updates/main amd64 Packages [377 kB]
Fetched 10.2 MB in 5s (1743 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
git is already the newest version (1:2.11.0-3+deb9u2).
make is already the newest version (4.1-9.1).
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.
Removing intermediate container 17fdae09cc87
 ---> 4a7d5f22b8ed
Step 4/10 : ADD . .
 ---> ded501c79341
Step 5/10 : RUN make build
 ---> Running in 595c0288c0c7
Removing intermediate container 595c0288c0c7
 ---> f6b116206d34
Step 6/10 : FROM debian:stretch
 ---> da653cee0545
Step 7/10 : COPY util/texlive.profile /
 ---> Using cache
 ---> aebce18e1dfc
Step 8/10 : RUN PACKAGES="wget libswitch-perl"     && apt-get update     && apt-get install -qq $PACKAGES --no-install-recommends     && wget -qO- http://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz |tar xz     && cd install-tl-*     && ./install-tl -profile /texlive.profile     && rm -rf install-tl-*     && apt-get remove --purge -qq $PACKAGES     && apt-get autoremove --purge -qq     && rm -rf /var/lib/apt/lists/*     && mkdir /var/tex
 ---> Using cache
 ---> 80cc7b2f02a7
Step 9/10 : COPY --from=build /go/bin/grafana-reporter /usr/local/bin
 ---> 17c7ba5d5d69
Step 10/10 : ENTRYPOINT [ "/usr/local/bin/grafana-reporter" ]
 ---> Running in 6f23865ce19d
Removing intermediate container 6f23865ce19d
 ---> d6105434f200
Successfully built d6105434f200
Successfully tagged grafana-reporter:0.6.2
```

**Comment by chiefy** [comment]
@IzakMarais bottom line is that if your github username has caps, it should follow in your go package namespacing, otherwise there's just gonna be problems.

**Comment by IzakMarais** [comment]
Aah I understand now. Since Travis isn't following the readme instructions (to use lowercase) but rather uses the correct case  that causes problems. Thanks for being patient with me. 

I agree with you that the correct fix is to change all references in the build scripts, readme and source code from `izakmarais` to `IzakMarais`. (I wish I had anticipated this when choosing my github user name). 

Would you be willing to make these changes in this PR?

**Comment by chiefy** [comment]
@IzakMarais there an option to change your GitHub name in settings. Not sure if that's the path you want to choose? Also what about TravisCI, can you change the command it uses to checkout from SCM?

**Comment by IzakMarais** [comment]
Thanks for the other options. Hmm changing my GitHub name sounds like that might cause unforeseen problems. I can't find a way to configure this in Travis, it seems like it automatically chooses the commands to run based on GitHub integration.  Hopefully I can find time in my work day to sort this mess out soon and get these quality PRs merged.

**Pull request merged**